olive/engine/engine.py

```diff
@@ -94,10 +95,10 @@ def __init__(
             self.evaluator = self._config.evaluator.create_evaluator()
 
         # dictionary of passes
+        self.pass_config = OrderedDict()
+
         # {"pass_name": {"pass": pass, "host": host, "evaluator": evaluator, "clean_run_cache": clean_run_cache}}
-        self.passes = {}
-        # list of pass names in the order they were registered
-        self.pass_order = []
+        self.passes = OrderedDict()
 
         self.footprints = Footprint()
```

Comment:
I think we would need to refresh this or create a new footprint object per hardware spec loop? Otherwise the models for different iterations are recorded into the same footprint object.

Reply:
@trajepl

Reply:
Maybe a field(accelerators) in footprints is better? Do you need me to implement it in the following PR?

Reply:
change to dict